Bio/phenotype/PhenMicro.py

```diff
@@ -0,0 +1,1126 @@
+# Copyright 2014 by Marco Galardini.  All rights reserved.
+# This code is part of the Biopython distribution and governed by its
+# license.  Please see the LICENSE file that should have been included
+# as part of this package.
+"""This module provides classes to work with Phenotype Microarray data.
+More information on the single plates can be found here: http://www.biolog.com/
+
+Classes:
+PlateRecord         Object that contain time course data on each well of the
+                    plate, as well as metadata (if any).
+WellRecord          Object that contains the time course data of a single well
+JsonWriter          Writer of PlateRecord objects in JSON format.
+
+Functions:
+JsonIterator        Incremental PM JSON parser, this is an iterator that returns
+                    PlateRecord objects.
+CsvIterator         Incremental PM CSV parser, this is an iterator that returns
+                    PlateRecord objects.
+_toOPM              Used internally by JsonWriter, converts PlateRecord objects
+                    in dictionaries ready to be serialized in JSON format."""
+
+import warnings
+import json
+import csv
+import numpy as np
+
+from Bio._py3k import basestring
+from Bio._py3k import _is_int_or_long
+from Bio import BiopythonParserWarning
+
+# Private csv headers - hardcoded because this are supposedly never changed
+_datafile = 'Data File'
+_plate = 'Plate Type'
+_strainType = 'Strain Type'
+_sample = 'Sample Number'
+_strainName = 'Strain Name'
+_strainNumber = 'Strain Number'
+_other = 'Other'
+_hour = 'Hour'
+_file = 'File'
+_position = 'Position'
+_setupTime = 'Setup Time'
+
+_platesPrefix = 'PM'
+_platesPrefixMammalian = 'PM-M'
+#
+
+# Json identifiers - hardcoded as they are set by the creators of opm
+_csvData = 'csv_data'
+_measurements = 'measurements'
+#
+
+
+class PlateRecord(object):
+    """PlateRecord object for storing Phenotype Microarray plates data.
+
+    A PlateRecord stores all the wells of a particular phenotype
+    Microarray plate, along with metadata (if any). The single wells can be
+    accessed calling their id as an index or iterating on the PlateRecord:
+
+    >>> from Bio import phenotype
+    >>> plate = phenotype.read("plate.csv", "pm-csv")
+    >>> well = plate['A05']
+    >>> for well in plate:
+    ...    print("%s" % well.id)
+    A01
```

Comment:
Minor observation: The Python documentation seems generally to include an extra (otherwise empty) line ``...`` to terminate the multi-line region, as you would see if typing this in at the python prompt.